crud/ProductManager/ProductManager/Classes/Manager.cs

```diff
@@ -86,19 +84,25 @@ public void UpdateProduct(Product updatedProduct)
             }
         }
 
-        public void DeleteProduct(int productId)
+        public void DeleteProduct(int productId, string filterStatus = null)
         {
             using (var connection = new NpgsqlConnection(_connectionString))
             {
                 connection.Open();
-                
                 var cmd = connection.CreateCommand();
-                cmd.CommandText = $"DELETE FROM public.\"Product\" WHERE \"Id\"={productId}";
+
+                if (!string.IsNullOrEmpty(filterStatus))
+                {
+                    cmd.CommandText = $"DELETE FROM public.\"Product\" WHERE \"Id\"={productId} AND \"Status\"='{filterStatus}'";
+                }
+                else
+                {
+                    cmd.CommandText = $"DELETE FROM public.\"Product\" WHERE \"Id\"={productId}";
```

Comment:
since you have duplicated query parts, it's better to move it to a separate string and than just add filterStatus part to that string if parameter is not null or empty.